wait-for-deployment

```diff
@@ -51,29 +52,41 @@ get_available_replicas() {
 get_deployment_jsonpath() {
   local -r jsonpath="$1"
 
-  kubectl get deployment "${deployment}" -o "jsonpath=${jsonpath}"
+  kubectl --namespace "${namespace}" get deployment "${deployment}" -o "jsonpath=${jsonpath}"
 }
 
 display_usage_and_exit() {
   echo "Usage: $(basename "$0") <deployment> <[timeout]>" >&2
```

Comment:
I noticed one last thing: could you update the help output please? It should now say

```bash
echo "Usage: $(basename "$0") [-n <namespace>] [-t <timeout>] <deployment>" >&2
```

Reply:
I've tried testing the script out with the changes that you suggested and I am having real trouble getting bash to play nicely with non-positional and positional arguments. As it stands, the script is broken. Either the positional arguments get read, or the non-positional arguments do; I either end up with blank or default values, and I can't work out how to count the number of non-positional arguments there are (even after `shift $((OPTIND-1))`)

It's a shame because the more I think about the changes you suggested, the more I like the idea of having just the required arguments being positional.

I've spent about half an hour trying to work out how to make it work, but I need to get on with my day job!

I'll have another go when I next get some time; for now - please don't accept my PR.